x-pack/plugins/security_solution/common/endpoint/service/artifacts/utils.ts

```diff
@@ -10,11 +10,19 @@ import type {
   CreateExceptionListItemSchema,
 } from '@kbn/securitysolution-io-ts-list-types';
 import { v4 as uuidv4 } from 'uuid';
-import { BY_POLICY_ARTIFACT_TAG_PREFIX, GLOBAL_ARTIFACT_TAG } from './constants';
+import type { EffectedPolicySelection } from '../../../../public/management/components/effected_policy_select';
+import type { PolicyData } from '../../types';
+import {
+  BY_POLICY_ARTIFACT_TAG_PREFIX,
+  FILTER_DESCENDENTS_OF_PROCESS_TAG,
+  GLOBAL_ARTIFACT_TAG,
+} from './constants';
+
+export type TagFilter = (tag: string) => boolean;
 
 const POLICY_ID_START_POSITION = BY_POLICY_ARTIFACT_TAG_PREFIX.length;
 
-export const isArtifactGlobal = (item: Pick<ExceptionListItemSchema, 'tags'>): boolean => {
+export const isArtifactGlobal = (item: Partial<Pick<ExceptionListItemSchema, 'tags'>>): boolean => {
```

Comment:
not necessarily empty object (i guess), but the `exception` prop in the forms, that is usually passed to `isArtifactGlobal` is either `ExceptionListItemSchema` or `CreateExceptionListItemSchema`, and in the latter `tags` is optional. because of this the function calls used type assertion: `isArtifactGlobal(item as ExceptionListItemSchema)`

so I wanted to remove the type assertions and instead prepare the utility function to receive the type as it is, and since it already handled `tags` as optional (`return (item.tags ?? []).find(...)`), i simply added the `Partial<>`

Reply:
got it. thanks for the response. I understand now the reason.